Framework/PythonInterface/mantid/plots/mantidaxes.py

```diff
@@ -27,8 +30,7 @@
 from mantid.plots import datafunctions, axesfunctions, axesfunctions3D
 from mantid.plots.legend import LegendProperties
 from mantid.plots.datafunctions import get_normalize_by_bin_width
-from mantid.plots.utility import (artists_hidden, autoscale_on_update,
-                                  legend_set_draggable, MantidAxType)
+from mantid.plots.utility import (artists_hidden, legend_set_draggable, MantidAxType)
```

Comment:
What impact will it have removing the autoscale_on_update?

Reply:
So `autoscale_on_update` was a decorator that caused issues with plotting with matplotlib 3.5, effectively, I had to go find wherever it was used and implement similar functionality for each of them. There should be no visual difference for this change.